Sahaj footer pt 90-106

sahajisy opened the pull request:
```
@FooterPage
Feature: Validate footer links on Footer Sections

#Testcase no: PT-090 to 95
Scenario: Validate Footer Links on Top Page

Then Check footer links on Top Page

#Testcase no: PT-122
Scenario: Check Links on Terms and Conditions Page

Then Check links in Terms of Use Page

#Testcase no: PT-102 to PT-105
Scenario: Validate Footer on Account and Services Page 
Then Check footer links on Login Page

#Testcase no: PT-096 to PT-101
Scenario: Validate Footer on User Top Page 

Given Login to Platform Portal
Then Check footer links on top page after login

#Testcase no: PT-102 to PT-105
Scenario: Validate Footer on Order Screen One

Given Login to Platform Portal
Then Check Footer Links on First Order Screen

#Testcase no: PT-102 to PT-105
Scenario: Validate Footer on Order Screen Two and Three

Given Login to Platform Portal
Then Check Footer Links on Second Order Screen
Then Check Footer Links on Third Order `Screen```
```

sahajisy committed:
Withdrawal and Footer Test Script Push

Pushed:
Withdrawal: Feature File, Step Definition File and Page Object File
Footer: Page Object File

sahajisy committed:
Merge branch 'main' into Sahaj_Footer_Withdrawal_PT-90_112

sahajisy committed:
Withdrawal Account and Footer Functions Push

Report:
Withdrawal: Feature File: Done, Step Definition: Done, PageObject: Done

Footer: Feature File: Done, Step Definition: Done, PageObject: Done
Note: Footer Function only execute footers without login.

sahajisy committed:
Footer Functions Update

Footer Page:
Total Test cases: 17
Test Cases Automated: 9/17
Pending Test Cases: 8
Page Object: Updated
Feature File: Updated
Step Definition: Updated

Note: Modified Footer Test Scripts.

sahajisy committed:
Withdrawal Removed

csonparote commented:
Please create exclusive tag name for scenario. will be useful for future tagging.
Please resolve error as per call

sahajisy committed:
FooterPage Code Change

@csonparote I have changed the tag for footer feature file.

sahajisy commented:
@csonparote I have added the Tag for Feature file also did the necessary changes for code to execute.

sahajisy committed:
Code Change: Feature File, Footer XPath

@csonparote I have change two xpaths of footer which were having the issue also done the changes on runner file as explained.

sahajisy committed:
Footer: Changes in Tab closing

@csonparote in this updated code, I have handled the close tab functionality.

sahajisy commented:
@csonparote Handled the close tab issue on footer links for external pages.

csonparote commented:
@sahaj-hirameki Please resolve conflicting files.

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/FooterPage.java:
```diff
@@ -9,141 +10,352 @@
 /**
- * @Author : Chetan Sonparote
- * @Date of Creation : 26 Jun 2021
+ * @Author : Chetan Sonparote, Sahaj Balgunde
```
Please create a new code block for authorship. dont merge into others

sahajisy replied:
Done the following change

csonparote replied:
Please add your seperate authorship block for code changes that u have done

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/FooterPage.java:
```diff
@@ -9,141 +10,352 @@
+		if (CommonFunctions.isElementClickable(termsOfUseLink)) {
+			termsOfUseLink.click();
+			log.info("Terms of Use Link is working");
+			Thread.sleep(3000);
```
Remove Thread.sleep mention from code.
Use wait from CommonFunctions for the same

sahajisy replied:
Removed Thread.sleep function

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/FooterPage.java:
```diff
@@ -9,141 +10,352 @@
 	}
 
-	public void clickTermsOfUseLink() {
+	public void accountServicesOrderFooter() throws Exception {
```
Please refactor/rename method in verb-noun format.
It is difficult to know what this method does

sahajisy replied:
Done changes as requested

csonparote replied:
Looks Good. thanks

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/FooterPage.java:
```diff
@@ -9,141 +10,352 @@
 
-	public void clickCramSchoolPitaLink() {
-		cramSchoolPitaLink.click();
+	public void AccountFooter() throws Exception {
```
Rename to verb-noun format

sahajisy replied:
Done changes as requested

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java:
```diff
@@ -0,0 +1,182 @@
+	}
+
+	@FindBy(xpath = "//a[contains(text(),'https://hiiragi.evidus.com/petz1/group/dfs/dantai/')]")
+	public WebElement casecTerms;
```
Plz mention type of element in name itself.
ex: casecTermsLink etc

sahajisy replied:
Will change some Web element names

sahajisy replied:
Terms and Conditions Pageobject is being handled by Sanae san.

csonparote replied:
Plz coordinate with Sanae san for the same.
Ensure that it is working fine in your code

sahajisy replied:
I have updated Terms and Conditions Code.

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java:
```diff
@@ -0,0 +1,182 @@
+	@FindBy(xpath = "//a[normalize-space()='http://www.ddai.info/optout']")
+	public WebElement ddaiOptout;
+
+	public void termsFunctions() throws Exception {
```
Rename to verb-noun format
Cant determine what this method does

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java:
```diff
@@ -0,0 +1,182 @@
+
+	}
+
+	public void termsLinks() throws Exception {
```
Rename to verb-noun format

csonparote commented on platform/src/test/java/platformfeatures/PF_Footer.feature:
```diff
@@ -0,0 +1,13 @@
+Feature: Validate footer links
+@FooterPage
```
Name is exclusive which is good.
Adding feature file name to tag would be better

csonparote commented on platform/src/test/java/platformfeatures/PF_Footer.feature:
```diff
@@ -0,0 +1,13 @@
+@FooterPage
+Scenario Outline: Validate Footer Links
+
+Given Navigate to URL and check footer links <browser>
```
Please break open browser into seperate step
Ex: Given Navigate to URL <browser>
When User clicks on footer links

sahajisy replied:
Done changes as requested

csonparote replied:
yup. We will edit it after merging to work on new runner

csonparote commented on platform/src/test/java/platformfeatures/PF_Footer.feature:
```diff
@@ -0,0 +1,13 @@
+
+Given Navigate to URL and check footer links <browser>
+#SAnd check footer links after login
+Then Close browser
```
Ex: 
Then validate links can be opened
And close browser

sahajisy replied:
Used And feature.

csonparote replied:
As per above comment, I cant determine what are you validating in Then step. Please add new step

sahajisy replied:
Updated Feature file.

sahajisy replied:
Updated Feature file.

csonparote commented on platform/src/test/java/platformstepdefinition/PF_Footer.java:
```diff
@@ -0,0 +1,51 @@
+@RunWith(Cucumber.class)
+public class PF_Footer extends NewBaseClass {
+
+	public WebDriver driver;
```
Replace with public WebDriver driver = ObjectHelper.driver;
Needed to retain browser instance

sahajisy replied:
Done changes as requested.

csonparote commented on platform/src/test/java/platformstepdefinition/PF_Footer.java:
```diff
@@ -0,0 +1,51 @@
+	public void navigate_to_URL_and_check_footer_links(String browser) throws Throwable {
+		driver = openbrowser(browser);
+		FooterPage footer = new FooterPage(driver);
+		footer.footerFunctions();
```
Rename to verb-non format
Cant determine what this method does

sahajisy replied:
Done changes as requested will confirm it again.

csonparote replied:
needs few edit for new runner. will discuss after code merge. Its good for now

csonparote commented on platform/src/test/java/platformstepdefinition/PF_Footer.java:
```diff
@@ -0,0 +1,51 @@
+		LoginPage loginToPlatform = new LoginPage(driver);
+		loginToPlatform.loginToPlatform();
+		FooterPage footer = new FooterPage(driver);
+		footer.afterLoginFooter();
```
Rename to verb-non format
Cant determine what this method does

sahajisy replied:
Done changes as requested.

csonparote replied:
Thanks

csonparote reviewed (changes requested):
Please do the needful

sahajisy committed:
Update PlatformTestRunners.java

sahajisy committed:
minor footer change

sahajisy committed:
Footer Test Case Update

Added Assert Function
Removed Thread Sleep.
Renamed Web Elements.

sahajisy committed:
Added After Login Top Page and Login Page Footer Validations.

Updated: After Login Top Page Footer Validation.
Login Page footer Validation

Added New Scenarios in Feature and Step Definition.

sahajisy commented:
Addded Some new features and step defs for Footer links on Top Page, Top Page after Login and Login Page.
Can you check it and try to execute it on you side?

sahajisy committed:
Update in Xpaths and Footer Links

sahajisy commented:
I am not able to handle authentication popup for jyuku pita, login and registration footer links. How can I handle it?

csonparote commented on platform/src/test/java/platformfeatures/PF_Footer.feature:
```diff
@@ -0,0 +1,15 @@
+Scenario Outline: Validate Footer Links
+
+Given Navigate to URL <browser>
+And Check footer links
```
Your validation starts here So this statement can be prefixed with Then.

csonparote commented on platform/src/test/java/platformfeatures/PF_Footer.feature:
```diff
@@ -0,0 +1,15 @@
+Given Navigate to URL <browser>
+And Check footer links
+Then Check footer links on Login Page
+Then Check footer links on top page after login
```
Multiple then statements not needed. You can prefix with And after one then statement.

sahajisy committed:
Update FooterPage.java

Added Excel Handler for assertion in FooterPage pageobject.

sahajisy committed:
Update FooterPage.java

sahajisy committed:
Merge branch 'main' into Sahaj_Footer_PT-90_106

sahajisy committed:
Create FooterPage_PT90_106.java

sahajisy committed:
Updated FooterPage file name.

Updated Footerpage file name to FooterPage_PT90_106 to avoid conflicts in Git.
Added FooterAssert sheet in NewTestData.xlsx.
Handled authentication popup for create account page title assertion.

sahajisy commented:
@csonparote san, 
Updated Footerpage file name to FooterPage_PT90_106 to avoid conflicts in Git.
Added FooterAssert sheet in NewTestData.xlsx.
Handled authentication popup for create account page title assertion.

sahajisy committed:
Updated Footer (Top Page, User Page, Order Page, Account and Services, Terms)

Feature: Validate footer links on Footer Sections

@FooterPage
#Testcase no: PT-090 to 95
Scenario: Validate Footer Links on Top Page

Then Check footer links on Top Page

@FooterPage
#Testcase no: PT-122
Scenario: Check Links on Terms and Conditions Page

Then Check links in Terms of Use Page

@FooterPage
#Testcase no: PT-102 to PT-105
Scenario: Validate Footer on Account and Services Page
Then Check footer links on Login Page

@FooterPage
#Testcase no: PT-096 to PT-101
Scenario: Validate Footer on User Top Page

Given Login to Platform Portal
Then Check footer links on top page after login

@FooterPage
#Testcase no: PT-102 to PT-105
Scenario: Validate Footer on Order Screen One

Given Login to Platform Portal
Then Check Footer Links on First Order Screen

@FooterPage
#Testcase no: PT-102 to PT-105
Scenario: Validate Footer on Order Screen Two and Three

Given Login to Platform Portal
Then Check Footer Links on Second Order Screen
Then Check Footer Links on Third Order Screen

sahajisy committed:
Updated Author Name on Terms and Condition PageObject, Updated Feature File

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java:
```diff
@@ -0,0 +1,723 @@
+		PageFactory.initElements(driver, this);
+	}
+
+	@FindBy(xpath = "//div[@class='footer-bt']//li[1]//a[1]")
```
Plz replace abs xpath with relative ones

sahajisy replied:
I checked the xpaths for Footer links but they have same class so It when I added those classes xpaths, it was not identifying. So had to take unique class from from webpage and had to add xpath from then on.

csonparote replied:
In case of design change, all these links will not be found as it is absolute path. Plz replace to relative xpath

sahajisy replied:
@csonparote san, I have updated the xpaths.

csonparote replied:
xpath not updated in this file

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java:
```diff
@@ -0,0 +1,723 @@
+	 * footerScbtLink.click(); }
+	 * 
+	 * }
+	 */
```
Please remove commented code, if you are never planning to use it

sahajisy replied:
Done the requested changes.

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java:
```diff
@@ -0,0 +1,371 @@
+	@FindBy(xpath = "//div[@class='ep-term__content']//div[15]//div[12]/a[1]")
+	public WebElement AboutadsLink;
+
+	@FindBy(xpath = "//div[@class='ep-term__content']//div[15]//div[13]/a[1]")
```
Plz replace absolute xpath with relative ones wherever applicable

sahajisy replied:
I checked the xpaths for terms and conditions links but they have same class so It when I added those classes xpaths, it was not identifying. So had to take unique class from from webpage and had to add xpath from then on.

sahajisy replied:
@csonparote san, I have updated the xpaths.

csonparote commented on platform/src/test/java/platformfeatures/PF_Footer.feature:
```diff
@@ -0,0 +1,35 @@
+@FooterPage
+Feature: Validate footer links on Footer Sections
```
Assign unique tag to feature file

csonparote commented on platform/src/test/java/platformstepdefinition/PF_Footer.java:
```diff
@@ -0,0 +1,86 @@
+	 * PendingException(); String environment = null; driver =
+	 * openBrowserstack(config, environment); }
+	 */
+
```
Remove commented code if not needed

sahajisy replied:
Done the requested changes.

csonparote reviewed (changes requested):
Please do the needful

sahajisy committed:
Merge branch 'main' into Sahaj_Footer_PT-90_106

sahajisy committed:
Updated Footer pageobject and step definition

sahajisy committed:
Merge branch 'main' into Sahaj_Footer_PT-90_106

sahajisy committed:
Merge branch 'main' into Sahaj_Footer_PT-90_106

csonparote reviewed (changes requested):
Plz do needful

sahajisy committed:
Updated Xpaths for Footer Section and Terms and conditions Page.

Updated Xpaths for Footer Section and Terms and conditions Page.

csonparote reviewed (approved):
Plz merge main into ur branch and check

csonparote reviewed (changes requested):
Please change xpath in platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java

sahajisy committed:
Update FooterPage_PT90_106.java

Updated xpaths for pageobject

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java:
```diff
@@ -0,0 +1,713 @@
+	@FindBy(xpath = "//a[@href='https://support.evidus.com/']")
+	public WebElement footerAccountQnA;
+
+	@FindBy(xpath = "//div[@class='ep-auth__footer']/ul[1]/li[2]//a[1]")
```
Plz replace this abs path

sahajisy replied:
@csonparote san, I have updated the xpaths.

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java:
```diff
@@ -0,0 +1,713 @@
+	@FindBy(xpath = "//a[@href='https://www.jiem.co.jp/privacy/']")
+	public WebElement footerAccountPrivacy;
+
+	@FindBy(xpath = "//div[@class='ep-auth__footer']/ul[1]/li[4]//a[1]")
```
Plz replace this abs path

sahajisy replied:
@csonparote san, I have updated the xpaths.

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java:
```diff
@@ -0,0 +1,713 @@
+	@FindBy(xpath = "//div[@class='ep-auth__footer']/ul[1]/li[4]//a[1]")
+	public WebElement footerAccountTax;
+
+	@FindBy(xpath = "//h3[contains(text(),'「スタギア」利用規約')]")
```
Plz replace with xpath without contains(text())

sahajisy replied:
Updated changes as requested

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java:
```diff
@@ -0,0 +1,713 @@
+	@FindBy(xpath = "//h3[contains(text(),'「スタギア」利用規約')]")
+	public WebElement footerTermsforAssert;
+
+	@FindBy(xpath = "//img[@alt='個人情報保護方針']")
```
PLz replace if possible. img tag is not advisable

sahajisy replied:
Updated Webelement name
Following Webelement is used for visibility.

sahajisy replied:
Updated xpaths as requested.

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java:
```diff
@@ -0,0 +1,713 @@
+	@FindBy(xpath = "//img[@alt='個人情報保護方針']")
+	public WebElement footerPrivacyforAssert;
+
+	@FindBy(xpath = "//h3[contains(text(),'「特定商取引に関する法律」に基づく表示')]")
```
Plz replace with xpath without contains(text())

sahajisy replied:
Updated changes as requested

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java:
```diff
@@ -0,0 +1,713 @@
+	@FindBy(xpath = "//h3[contains(text(),'「特定商取引に関する法律」に基づく表示')]")
+	public WebElement footerTaxforAssert;
+
+	@FindBy(xpath = "//img[@alt='スタギアサポートセンター ヘルプセンターのホームページ']")
```
Plz replace with xpath without contains(text())

sahajisy replied:
Updated changes as requested

csonparote commented on platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java:
```diff
@@ -0,0 +1,713 @@
+	@FindBy(xpath = "//img[@alt='スタギアサポートセンター ヘルプセンターのホームページ']")
+	public WebElement footerQandAforAssert;
+
+	@FindBy(xpath = "//img[@alt='英検S-CBTを平日に受験しよう！']")
```
Plz replace img tag with alternate xpath

sahajisy replied:
Updated Webelement name
Following Webelement is used for visibility.

sahajisy replied:
Updated xpaths as requested.

csonparote reviewed (changes requested):
Plz make changes

sahajisy committed:
Updated xpaths and functions on Footer Test Scripts

sahajisy committed:
Updated xpaths for visibility webelements.

csonparote reviewed (approved):
Plz merge main into this branch and check

sahajisy committed:
Merge branch 'main' into Sahaj_Footer_PT-90_106

sahajisy committed:
Updated FooterPage PageObject, Step Definition and Feature File
